commands/wiki.js

```diff
@@ -1,48 +1,61 @@
-  request = require("request-promise-native"),
-  u = require("../utils/utils");
+const Augur = require("augurbot")
+  u = require("../utils/utils"),
+  axios = require('axios');
+
+async function getPage(page) {
+  return new Promise((fulfill, reject) => {
+    let url = "https://wiki.ldsgamers.com/api/" + page;
+    let token = Module.config.api.wiki;
+    axios.request({
+      url,
+      method: "get",
+      headers: { "Authorization": `Token ${token.tokenId}:${token.tokenSecret}` },
+      data
+    }).then(fulfill, reject);
+  })
+}
+
+function levenshteinDistance(s, t) {
+  if (!s.length) return t.length;
+  if (!t.length) return s.length;
+
+  return Math.min(
+    levenshteinDistance(s.substr(1), t) + 1,
+    levenshteinDistance(t.substr(1), s) + 1,
+    levenshteinDistance(s.substr(1), t.substr(1)) + (s[0] !== t[0] ? 1 : 0)
+  ) + 1;
+}
 
 const Module = new Augur.Module()
-.addCommand({name: "wiki",
-  description: "Search Wikipedia for a term.",
+.addCommand({
+  name: "wiki",
+  description: "Search the LDSG Wiki for a term.",
   syntax: "Term",
-  permissions: (msg) => (!msg.guild || (msg.channel.permissionsFor(msg.member).has(["EMBED_LINKS", "ATTACH_FILES"]) && msg.channel.permissionsFor(msg.client.user).has("ATTACH_FILES"))),
   process: async (msg, suffix) => {
-    try {
-      if (!suffix) return msg.reply("you need to tell me what to search!").then(u.clean);
-      const pf = new profanityFilter();
-      if (pf.scan(suffix.toLowerCase()).length == 0) {
-        let wiki = JSON.parse(await request(`https://en.wikipedia.org/w/api.php?action=opensearch&format=json&redirects=resolve&search=${encodeURIComponent(suffix)}`));
-
-        if (wiki && wiki[1].length > 0) {
-          let embed = u.embed()
-          .setTitle(wiki[1][0])
-          .setDescription(wiki[2][0])
-          .setURL(wiki[3][0])
-          .setAuthor("Wikipedia", "https://upload.wikimedia.org/wikipedia/commons/thumb/8/80/Wikipedia-logo-v2.svg/103px-Wikipedia-logo-v2.svg.png");
-
-          if (wiki[2][0].endsWith("may refer to:")) {
-            for (let i = 1; i < wiki[1].length; i++) {
-              embed.addField(wiki[1][i], wiki[2][i] + `\n[[Link]](${wiki[3][i]})`);
-            }
-          } else {
-            let body = await request(wiki[3][0]);
-            let $ = cheerio.load(body);
-
-            let img = $("table.infobox img");
-            if (img.length > 0) {
-              embed.setThumbnail("https:" + img.first().attr("src"));
-            }
-          }
-          msg.channel.send({embed});
-        } else {
-          msg.reply(`I couldn't find a result for \`${suffix}\` on <https://en.wikipedia.org>.`);
-        }
-      } else msg.reply("I'm not going to search for that. :rolling_eyes:").then(u.clean);
-    } catch(e) { u.errorHandler(e, msg); }
+    let items = (await getPage("shelves").catch((err) => { return { data: [] }; })).data
+      .concat((await getPage("books").catch((err) => { return { data: [] }; })).data);
```

Comment:
While I *think* this is technically functional, it's weird and clunky to read. (Yes, I know you based it on something I sent earlier.) Let's keep it clean.
```suggestion
    let shelves = (await getPage("shelves").catch(u.noop))?.data || [];
    let books = (await getPage("books").catch(u.noop))?.data || [];
    let items = shelves.concat(books);
```